File: backend/app/schemas/customer.py
from typing import Optional
from pydantic import BaseModel, EmailStr
from uuid import UUID

class CustomerBase(BaseModel):
    name: str
    ruc: Optional[str] = None
    dv: Optional[str] = None
    email: Optional[EmailStr] = None
    phone: Optional[str] = None
    address: Optional[str] = None

class CustomerCreate(CustomerBase):
    pass

class CustomerUpdate(CustomerBase):
    name: Optional[str] = None

class CustomerSchema(CustomerBase):
    id: UUID

    class Config:
        from_attributes = True

File: frontend/Models/Customer.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;

namespace frontend.Models
{
    public class Customer
    {
        [JsonPropertyName("id")]
        public Guid Id { get; set; }
        [JsonPropertyName("name")]
        public string Name { get; set; } = string.Empty;
        [JsonPropertyName("ruc")]
        public string? Ruc { get; set; }
        [JsonPropertyName("dv")]
        public string? Dv { get; set; }
        [JsonPropertyName("email")]
        public string? Email { get; set; }
        [JsonPropertyName("phone")]
        public string? Phone { get; set; }
        [JsonPropertyName("address")]
        public string? Address { get; set; }
    }

    public class CustomerCreate
    {
        [Required(ErrorMessage = "El nombre es requerido")]
        [JsonPropertyName("name")]
        public string Name { get; set; } = string.Empty;
        [JsonPropertyName("ruc")]
        public string? Ruc { get; set; }
        [JsonPropertyName("dv")]
        public string? Dv { get; set; }
        [EmailAddress(ErrorMessage = "Email inválido")]
        [JsonPropertyName("email")]
        public string? Email { get; set; }
        [JsonPropertyName("phone")]
        public string? Phone { get; set; }
        [JsonPropertyName("address")]
        public string? Address { get; set; }
    }

    public class CustomerUpdate : CustomerCreate
    {
    }
}
